pkg/volume/metadata/metadata.go

```diff
@@ -0,0 +1,309 @@
+		var data string
+		var err error
+		if data, err = fieldpath.ExtractFieldPathAsString(m.pod, fieldReference); err != nil {
+			glog.Error(err)
+			errlist = append(errlist, err)
+		}
+		if isFileToGenerate(path.Join(m.GetPath(), fileName), sortLines(data)) {
+			return true, utilErrors.NewAggregate(errlist)
+		}
+	}
+	return false, utilErrors.NewAggregate(errlist)
+}
+
+// isFileToGenerate compares actual file with the new values. If
+// different (or the file does not exist) return true
+func isFileToGenerate(fileName, values string) bool {
+	if _, err := os.Lstat(fileName); os.IsNotExist(err) {
+		return true
+	}
+	return readFile(fileName) != values
+}
+
+const currentDir = ".current"
+const currentTmpDir = ".current_tmp"
+
+// writeData writes requested metadata in specified files.
+//
+// The file visible in this volume are symlinks to files in the '.current'
+// directory. Actual files are stored in an hidden timestamped directory which is
+// symlinked to by '.current'. The timestamped directory and '.current' symlink
+// are created in the plugin root dir.  This scheme allows the files to be
+// atomically updated by changing the target of the '.current' symlink.  When new
+// data is available:
+//
+// 1.  A new timestamped dir is created by writeDataInTmpDir
+// 2.  Symlinks for new files are created if needed
+// 3.  The previous timestamped directory is detected reading the '.current' symlink
+// 4.  In case no symlink exists then it's created and func returns (first execution)
+// 5.  In case symlink exists a new temporary symlink is created .current_tmp
+// 6.  .current_tmp is renamed to .current
+// 7.  The previous timestamped directory is removed
+func (m *metadataVolume) writeData() error {
+	var tempDir string
+	var err error
+	tempDir, err = m.writeDataInTmpDir()
+	if err != nil {
+		glog.Error(err)
+		return err
+	}
+	// update symbolic links for relative paths
+	if err = m.updateSymlinks(); err != nil {
+		os.RemoveAll(tempDir)
+		glog.Error(err)
+		return err
+	}
+
+	_, tempDirBaseName := filepath.Split(tempDir)
+	var oldTmpDirectory string
+	oldTmpDirectory, err = os.Readlink(path.Join(m.GetPath(), currentDir))
+	if err != nil {
+		// no link to currentDir so creates it and return
+		err = os.Symlink(tempDirBaseName, path.Join(m.GetPath(), currentDir))
+		return err
+	} else {
+		// nominal case: link to currentDir exists create the link to currentTmpDir
+		if err = os.Symlink(tempDirBaseName, path.Join(m.GetPath(), currentTmpDir)); err != nil {
+			glog.Error(err)
+			return err
+		}
+	}
+	// Rename the symolic link currentTmpDir to currentDir
+	if err = os.Rename(path.Join(m.GetPath(), currentTmpDir), path.Join(m.GetPath(), currentDir)); err != nil {
+		// in case of error remove latest data and currentTmpDir
+		os.Remove(path.Join(m.GetPath(), currentTmpDir))
+		os.RemoveAll(tempDir)
+		glog.Error(err)
+		return err
+	}
+	if len(oldTmpDirectory) > 0 { // Remove the old tmpDirectory
```

Comment:
likewise, I think `oldTmpDirectory` should become `oldTimestampDirectory` or some shortened form of that.